cid/common.py

```diff
@@ -1239,23 +1248,70 @@ def get_data_from_definition(self, asset_type, definition):
             ).decode('utf-8')
             if file_name.endswith('.json') or file_name.endswith('.jsn'):
                 data = json.loads(text)
+            elif file_name.endswith('.yaml') or file_name.endswith('.yml'):
+                data = yaml.safe_load(text)
             else:
                 data = text
         if data is None:
             raise CidCritical(f"Error: definition is broken. Cannot find data for {repr(definition)}. Check resources file.")
         return data
 
+
     def create_datasource(self, datasource_id) -> str:
+        """Create datasource with given id
+        uses parameters: 'quicksight-datasource-role-arn', 'quicksight-datasource-role'
+        """
         role_arn = get_parameters().get('quicksight-datasource-role-arn')
         if not role_arn:
             role_name = get_parameters().get('quicksight-datasource-role')
             if role_name:
                 role_arn = f'arn:aws:iam::{self.base.account_id}:role/{role_name}'
+
+        if not role_arn:
+            quicksight_trusted_roles = list(self.iam.iterate_role_names(search="Roles[?AssumeRolePolicyDocument.Statement[?Principal.Service=='quicksight.amazonaws.com']].Arn"))
+            quicksight_trusted_roles = [role for role in quicksight_trusted_roles if role.split('/')[-1] not in ('aws-quicksight-secretsmanager-role-v0')] # filter out irrelevant roles
+            # TODO: filter only roles with Athena and S3 policies
+
+            cid_role_name = 'CidCmdQuicksightDatasetRole'
```

Comment:
Let's align it with cfn role
```suggestion
            cid_role_name = 'CidQuickSightDataSourceRole'
```